presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -936,6 +960,84 @@ public synchronized void rollback()
         }
     }
 
+    public void beginQuery(ConnectorSession session)
+    {
+        String queryId = session.getQueryId();
+        HiveIdentity identity = new HiveIdentity(session);
+
+        synchronized (this) {
+            checkState(
+                    !currentQueryId.isPresent() && !hiveTransactionSupplier.isPresent(),
+                    "Query already begun: %s while starting query %s",
+                    currentQueryId,
+                    queryId);
+            currentQueryId = Optional.of(queryId);
+
+            hiveTransactionSupplier = Optional.of(() -> {
+                long heartbeatInterval = configuredTransactionHeartbeatInterval
+                        .map(Duration::toMillis)
+                        .orElseGet(this::getServerExpectedHeartbeatIntervalMillis);
+                long transactionId = delegate.openTransaction(identity);
+                log.debug("Using hive transaction %s for query %s", transactionId, queryId);
+
+                ScheduledFuture<?> heartbeatTask = heartbeatExecutor.scheduleAtFixedRate(
+                        () -> delegate.sendTransactionHeartbeat(identity, transactionId),
+                        0,
+                        heartbeatInterval,
+                        MILLISECONDS);
+
+                return new HiveTransaction(identity, queryId, transactionId, heartbeatTask);
+            });
+        }
+    }
+
+    private long getServerExpectedHeartbeatIntervalMillis()
+    {
+        String hiveServerTransactionTimeout = delegate.getConfigValue(TXN_TIMEOUT.toString()).orElse(TXN_TIMEOUT.getDefaultVal().toString());
+        Configuration configuration = new Configuration();
```

Comment:
```suggestion
        Configuration configuration = new Configuration(false);
```

otherwise this loads you-don't-know-what config files